### ISSUE TITLE
Narrowing conversion inside { } is ill-formed

### ISSUE THREAD

**Issue by DavidEGrayson** [pololu/pololu-led-strip-arduino#4]
The way we initialize the color structs in the demo code is actually "ill-formed" in C++11.  For example:

``` c++
(rgb_color){ 0, 255 - x, 0 };
```

The problem is that the middle field is a uint8_t, but the value we are trying to stick into it is actually an int because of integer promotion.  I found out because I tried to compile code like that in Atmel Studio, which is using a pretty modern version of the GCC C++ compiler and got warnings like this:

```
narrowing conversion of '(255 - ((int)x))' from 'int' to 'unsigned char' inside { } is ill-formed in C++11 [-Wnarrowing]
```

When the Arduino IDE starts using a modern compiler and people get this warning, we should introduce a macro or inline function like this:

``` c++
#define color(r, g, b) ((rgb_color){ (uint8_t)(r), (uint8_t)(g), (uint8_t)(b) })
```

It is OK to have a narrowing happen in a normal assignment, like c = a + b, but if narrowing happens during an initializer they think that is special and deserves a warning.

**Comment by MuneshThakur** [comment]
Hello EveryOne
                            I want to On/Off and change the color if this Rgb Dtrip using Android tablet with Arduino uno and these all thin are connected via Usb Cable .i will wait your reply
Thanks & Regard
Munesh parmar

**Comment by MuneshThakur** [comment]
Hello Every One
                            I want to On/Off and change  color of  this Rgb strip using Android tablet with Arduino uno and these all think are connected via USB Cable.so can you give me a any suggestion i will wait your reply
Thanks & Regards
Munesh parmar

**Comment by ryantm** [comment]
Hi, @MuneshThakur.

As @DavidEGrayson mentioned in his reply to you in #9, this is not the appropriate place to ask for this kind of help. Please stop posting asking for help here and try on [our forum](http://forum.pololu.com/), where people other than the library developers are more likely to read it.

Ryan

**Comment by DavidEGrayson** [comment]
This is fixed in version 4.2.0 with the new constructors for rgb_color.
